src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -0,0 +1,216 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/** */
+public final class SearchServletTest {
+
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private SearchServlet searchObject;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    searchObject = new SearchServlet();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+
+  @Test
+  /* Ensure the correct filters are set when all search parameters are specified. */
+  public void MakeFilters_AllParamsSet() {
+    request = createRequest(request, "CS 105", "Smith", "Spring 2020", "1,2,3");
+
+    List<Filter> expectedFilters = new ArrayList<>();
+    List<Integer> units = new ArrayList<Integer>(Arrays.asList(1, 2, 3));
+
+    expectedFilters.add(new FilterPredicate("name", FilterOperator.EQUAL, "CS 105"));
+    expectedFilters.add(new FilterPredicate("professor", FilterOperator.EQUAL, "Smith"));
+    expectedFilters.add(new FilterPredicate("term", FilterOperator.EQUAL, "Spring 2020"));
+    expectedFilters.add(new FilterPredicate("units", FilterOperator.IN, units));
+
+    List<Filter> filters = searchObject.getFilters(request);
+
+    assertEquals(filters, expectedFilters);
+  }
+
+  @Test
+  /* Ensure no filters are made given no search parameters are specified. */
+  public void MakeFilters_NoParamsSet() {
+
+    // TODO: Find a better way to replicate null parameters -
+    // getting NullPointerException when trying other methods.
+    request = createRequest(request, "", "", "select", "");
+
+    List<Filter> expectedFilters = new ArrayList<>();
+
+    List<Filter> filters = searchObject.getFilters(request);
+
+    assertEquals(filters, expectedFilters);
+  }
+
+  @Test
+  /* Ensure the correct results are retrieved from Datastore given that all filters are set. */
+  public void GetResults_AllFiltersSet() {
+
+    request = createRequest(request, "CS 105", "Smith", "Spring 2020", "1,2,3");
+    List<Entity> expectedResults = new ArrayList<>();
+
+    addCourseEntityWithList("CS 105", "Smith", "Spring 2020", 1, expectedResults, true);
+    addCourseEntityWithList("CS 105", "Smith", "Spring 2020", 4, expectedResults, false);
+
+    List<Filter> filters = searchObject.getFilters(request); // Tested in makeFiltersAllParams.
+    List<Entity> results = searchObject.getResults(filters);
+
+    assertEquals(results, expectedResults);
+  }
+
+  @Test
+  /* Ensure the correct results are retrieved from Datastore given no search filters. */
+  public void GetResults_NoFiltersSet() {
+
+    request = createRequest(request, "", "", "select", "");
+    List<Entity> expectedResults = new ArrayList<>();
+
+    addCourseEntityWithList("CS 105", "Smith", "Spring 2020", 1, expectedResults, true);
```

Comment:
What does this variable name mean here?

Reply:
Sorry, what variable? If you look at the function declaration I think it's name, professor, term, units, array to potentially add to, and whether to add.